### PR TITLE
Automated test for saltutil.kill_job

### LINKED ISSUE #34143: saltutil.kill_job test
```Preconditions
"# salt centos-7-main -v cmd.run 'sleep 1024' python_shell=True &
Executing job with jid <JID>
-------------------------------------------
Steps
salt centos-7-main saltutil.kill_job <JID>
Expected Result
centos-7-main:
    Signal 9 sent to job <JID> at pid <PID>

centos-7-main:
    Minion did not return. [Not connected]

[1]+  Done                    salt centos-7-main -v cmd.run 'sleep 1024' python_shell=True"
```

### PR DESCRIPTION
**Pull request by cachedout**
This adds an automated test for `saltutil.kill job`
that starts a job in async mode, then tests to see if
it is running and then kills it and ensures it is not running.

Also adds the ability for salt modules to be called asynchronously
from the test suite.

Closes #34143

### CODE REVIEW

File: tests/integration/__init__.py [hunk 1]
```diff
@@ -1091,17 +1091,23 @@ def minion_run(self, _function, *args, **kw):
...
-        )
+        if async:
+            orig = self.client.cmd_async(
+                minion_tgt, function, arg, timeout=timeout, kwarg=kwargs
+            )
+            return orig
+        else:
+            orig = self.client.cmd(
+                minion_tgt, function, arg, timeout=timeout, kwarg=kwargs
+            )
```
Comment (s0undt3ch) [inline tests/integration/__init__.py]: You return orig above but not here. Return orig outside the if/else block.

Comment (s0undt3ch) [inline tests/integration/__init__.py]: Or I saw this wrong?

Comment (cachedout) [inline tests/integration/__init__.py]: This is intentional. We don't want to continue processing at this point. We have everything we need. However, the sync version needs to go on for additional checks.

Comment (s0undt3ch) [inline tests/integration/__init__.py]: :+1:

**Comment by rallytime** [comment]
Hrm. Looks like this new test is failing. Mind taking another look @cachedout?

**Comment by rallytime** [comment]
@cachedout This test is still failing with the new change (and some pesky whitespace :) )

**Comment by cachedout** [comment]
Guh. That's...odd.

**Pull request closed without merging**